src/com/amazon/ion/impl/bin/utf8/Poolable.java

```diff
@@ -0,0 +1,30 @@
+package com.amazon.ion.impl.bin.utf8;
+
+import java.io.Closeable;
+
+/**
+ * Base class for types that may be pooled.
+ * @param <T> the concrete type.
+ */
+abstract class Poolable<T extends Poolable<T>> implements Closeable {
```

Comment:
This type bound is to ensure that only subclasses of `Poolable` can implement `Poolable`, right? Which provides the guarantee that you'll get instantiated with reference to an appropriate `Pool`?

Reply:
Yeah, it allows us to guarantee that we're getting a `Pool<T>` (i.e. a `Pool` for this type of `Poolable`) in the constructor.